### ISSUE TITLE
"Identifier * has already been cleared" when using a spec definition as record key

### ISSUE THREAD

**Issue by martindines** [iyegoroff/spectypes#2]
Hi @iyegoroff

I am trying to use a validator to validate the key of a record, but am hitting a compilation error (see below).
My use case is to have one definition of a key validator (in the below example, UsernameSpec) which can be used in other validators (UsersSpec record, FriendsSpec array, etc). Before I dig in to the transformed code, I am wondering if this is something you have come across and have a solution? Thanks

```
import {struct, record, string} from "spectypes";

const UsernameSpec = string;

const UserSpec = struct({
    username: UsernameSpec
});

export const UsersSpec = record(UsernameSpec, UserSpec);
export const FriendsSpec = array(UsernameSpec);
```

Results in the follow error:

```
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Volumes/CS/projects/___/application/resources/assets/app/src/___/___/___.schema.ts: Identifier 'result_key' has already been declared. (1:761)
    at instantiate (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:67:32)
    at constructor (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:364:12)
    at Parser.raise (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:3363:19)
    at ScopeHandler.checkRedeclarationInScope (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:1587:19)
    at ScopeHandler.declareName (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:1558:12)
    at Parser.declareNameFromIdentifier (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:10767:16)
    at Parser.checkIdentifier (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:10763:12)
    at Parser.checkLVal (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:10697:12)
    at Parser.parseVarId (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13671:10)
    at Parser.parseVar (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13649:12)
    at Parser.parseVarStatement (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13469:10)
    at Parser.parseStatementContent (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13050:23)
    at Parser.parseStatementLike (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12963:17)
    at Parser.parseStatementListItem (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12948:17)
    at Parser.parseBlockOrModuleBlockBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13569:61)
    at Parser.parseBlockBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13561:10)
    at Parser.parseBlock (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13549:10)
    at Parser.parseStatementContent (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13057:21)
    at Parser.parseStatementLike (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12963:17)
    at Parser.parseStatement (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12955:17)
    at /Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13638:10
    at Parser.withSmartMixTopicForbiddingContext (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12628:14)
    at Parser.parseForIn (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13637:10)
    at Parser.parseForStatement (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13295:21)
    at Parser.parseStatementContent (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12982:21)
    at Parser.parseStatementLike (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12963:17)
    at Parser.parseStatementListItem (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12948:17)
    at Parser.parseBlockOrModuleBlockBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13569:61)
    at Parser.parseBlockBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13561:10)
    at Parser.parseBlock (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13549:10)
    at Parser.parseStatementContent (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13057:21)
    at Parser.parseStatementLike (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12963:17)
    at Parser.parseStatementOrFunctionDeclaration (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12951:17)
    at Parser.parseIfStatement (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13350:42)
    at Parser.parseStatementContent (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13001:21)
    at Parser.parseStatementLike (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12963:17)
    at Parser.parseStatementListItem (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12948:17)
    at Parser.parseBlockOrModuleBlockBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13569:61)
    at Parser.parseBlockBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13561:10)
    at Parser.parseBlock (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:13549:10)
    at Parser.parseFunctionBody (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12298:24)
    at Parser.parseArrowExpression (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:12271:10)
    at Parser.parseParenAndDistinguishExpression (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:11840:12)
    at Parser.parseExprAtom (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:11457:23)
    at Parser.parseExprSubscripts (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:11182:23)
    at Parser.parseUpdate (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:11164:21)
    at Parser.parseMaybeUnary (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:11138:23)
    at Parser.parseMaybeUnaryOrPrivate (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:10967:61)
    at Parser.parseExprOps (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:10973:23)
    at Parser.parseMaybeConditional (/Volumes/CS/projects/___/application/resources/assets/app/node_modules/@babel/parser/lib/index.js:10948:23)
```

**Comment by iyegoroff** [comment]
Hi @martindines 

That's a bug! I'll try to fix it in a week or so.

**Comment by iyegoroff** [comment]
Fixed in v2.1.8

**Comment by martindines** [comment]
Many thanks! :-)
